dashboard/app/models/level.rb

```diff
@@ -75,8 +75,9 @@ def level_num_custom?
   end
 
   def self.load_custom_levels
+    levels = Level.includes(:game).all.index_by(&:name)
```

Comment:
hm, I'm noticing that you've changed everything in to a class methods -- however you are now passing around some state. It makes sense that this state does not belong in a Level instance... I wonder if it would be cleaner to pull all of these methods into a eg. CustomLevelLoader class with the level by name index as an instance variable. 

Optional :)